libs/common/utils/object/src/lib/common-utils-object.ts

```diff
@@ -9,18 +9,14 @@
  * @param lookup Source object
  * @param property Dot notation string representing the location of the property
  */
-export function getPropertyValue(lookup: object, property: string): any {
+export function getPropertyValue<T>(lookup: object, property: string): T {
   if (!lookup || !property) {
     return undefined;
   }
 
   const path = property.split('.');
   let value = JSON.parse(JSON.stringify(lookup));
 
-  if (path.length === 0) {
-    return undefined;
-  }
```

Comment:
According to [the spec](http://www.ecma-international.org/ecma-262/5.1/#sec-15.5.4.14) it is impossible for `String.split` to return an array with less than 1 element. In the case that the pattern (in this case: `'.'`) is not found then the entire string is returned.

Reply:
Good catch.